x-pack/plugins/monitoring/common/runtime_types.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { fold } from 'fp-ts/lib/Either';
+import { pipe } from 'fp-ts/lib/pipeable';
+import { Context, Errors, IntersectionType, Type, UnionType, ValidationError } from 'io-ts';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
```

Comment:
This caught my attention. If we're adding new code with eslint disables maybe we should file an issue about the lint configuration or exporting of `RouteValidationFunction`?

Reply:
Yeah, that's fair. I had copied this verbatim from infra, but it looks like we can import `RouteValidationFunction` from `src/core/server` now without any linting overrides, so I'll change the import here and in infra.

Reply:
Okay, my bad, I wasn't getting any linting errors locally so I thought this change was okay. But it triggered a failure on CI. The issue is we're importing into `common`, if it was `server` it would be fine. 

I don't necessarily think changing the overall platform linting configuration is a good idea as having these `server` and `public` import rules is good and enforces better behaviour. But, we do have these `common` directories (and they are super useful for shared types). I don't think this should block anything, but I can try and engage with someone from Kibana at some point regarding this pattern.

Reply:
I resolved this by moving the server-side validation function code to `server/lib/create_validation_function.ts`. That way common runtime type helper code remains in `common` while `server` code only lives in `server`. We should apply that to the `infra` plugin too, but I didn't want to bloat this PR.